### PR TITLE
[MOD-119][MOD-103][MOD-101] Refactor permission checks

### PR DESCRIPTION
**Pull request by laurenbarker**
## Changes
* [MOD-119] Remove `not-authenticated` route and display not authenticated page on index
    * Checks `session.isAuthenticated` in `application.js`
* [MOD-103][MOD-101] Check `canViewReviews` in `application.js`
    * Keep `view_submissions` check in `provider.js`
    * Keep `set_up_moderation` check in `setup.js`

### CODE REVIEW

File: app/routes/application.js [hunk 1]
```diff
@@ -32,5 +35,16 @@ export default Ember.Route.extend(OSFAgnosticAuthRouteMixin, {
         if (locale) {
             this.set('i18n.locale', locale);
         }
+
+        // Check permissions
+        if (!this.get('session.isAuthenticated')) {
+            this.replaceWith('index');
+        } else {
+            this.get('currentUser.user').then((user) => {
```
Comment (aaxelb) [inline app/routes/application.js]: Ooh, I think *this* is when we want to return something from a model hook. The transition should pause until this promise resolves.

File: app/routes/application.js [hunk 2]
```diff
@@ -32,5 +35,16 @@ export default Ember.Route.extend(OSFAgnosticAuthRouteMixin, {
         if (locale) {
             this.set('i18n.locale', locale);
         }
+
+        // Check permissions
```
Comment (aaxelb) [inline app/routes/application.js:39]: This seems like it'd make more sense in `beforeModel`, so it won't try fetching all the models before redirecting to `index`.

**Review by aaxelb — CHANGES_REQUESTED** [review]
Nice 🍖 

Just a couple small things

File: app/routes/application.js (hunk 2 above)
Comment (laurenbarker) [reply app/routes/application.js:39]: I had it in `beforeModel` originally but session wasn't defined yet so it always said the user wasn't logged in.

File: app/routes/application.js (hunk 1 above)
Comment (laurenbarker) [reply app/routes/application.js]: 👍  will try that

File: app/routes/application.js (hunk 2 above)
Comment (aaxelb) [reply app/routes/application.js:39]: Aw, that's annoying

**Comment by laurenbarker** [comment]
@aaxelb, addressed 👗

**Review by aaxelb — APPROVED** [review]

**Pull request merged**